src/qt/guiutil.cpp

```diff
@@ -1057,14 +1058,49 @@ void loadStyleSheet(QWidget* widget, bool fDebugWidget)
                 return false;
             }
 
+            std::string platformName = gArgs.GetArg("-uiplatform", BitcoinGUI::DEFAULT_UIPLATFORM);
             stylesheet = std::make_unique<QString>();
 
             for (const auto& file : vecFiles) {
                 QFile qFile(file);
                 if (!qFile.open(QFile::ReadOnly)) {
                     throw std::runtime_error(strprintf("%s: Failed to open file: %s", __func__, file.toStdString()));
                 }
-                stylesheet->append(QLatin1String(qFile.readAll()));
+
+                QString strStyle = QLatin1String(qFile.readAll());
+                // Process all <os=...></os> groups in the stylesheet first
+                QRegularExpressionMatch osStyleMatch;
+                QRegularExpression osStyleExp(
+                        "^"
+                        "(<os=(?:'|\").+(?:'|\")>)" // group 1
+                        "((?:.|\n)+?)"              // group 2
+                        "(</os>?)"                  // group 3
+                        "$");
+                osStyleExp.setPatternOptions(QRegularExpression::MultilineOption);
+                QRegularExpressionMatchIterator it = osStyleExp.globalMatch(strStyle);
+
+                // For all <os=...></os> sections
+                while (it.hasNext() && (osStyleMatch = it.next()).isValid()) {
+                    QStringList listMatches = osStyleMatch.capturedTexts();
+
+                    // Full match + 3 group matches
+                    if (listMatches.size() % 4) {
+                        throw std::runtime_error(strprintf("%s: Invalid <os=...></os> section in file %s", __func__, file.toStdString()));
+                    }
+
+                    for (int i = 0; i < listMatches.size(); i += 4) {
```

Comment:
It probably does matter, but QStringList seems like it should basically be a vector, but fancy, and you "should" use size_t with vectors, so maybe you should use size_t here?
```suggestion
                    for (size_t i = 0; i < listMatches.size(); i += 4) {
```

Reply:
It should be `int` actually https://doc.qt.io/qt-5/qlist.html#size

Reply:
Good that at least one of three looks at the docs hehe, reverted!